src/app/routes/regex/index.js

```diff
@@ -11,13 +12,18 @@ const assetUriRegex = '[a-z-_]{0,}[0-9]{8,}';
 
 const variantRegex = '/simp|/trad|/cyr|/lat';
 
-export const articleRegexPath = `/:service(${serviceRegex})/articles/:id(${idRegex}):variant(${variantRegex})?:amp(${ampRegex})?`;
+const articlePathnameRegex = Object.keys(pathnames).reduce((regex, service) => {
+  const pathname = pathnames[service].articles;
+  return `${regex}|${pathname}`;
+}, '');
+
+export const articleRegexPath = `/:service(${serviceRegex})/:pathname(${articlePathnameRegex})/:id(${idRegex}):variant(${variantRegex})?:amp(${ampRegex})?`;
 
 export const articleDataRegexPath = `${articleRegexPath}.json`;
 
-export const articleSwRegexPath = `/:service(${serviceRegex})/articles/sw.js`;
+export const articleSwRegexPath = `/:service(${serviceRegex})/:pathname(${articlePathnameRegex})/sw.js`;
```

Comment:
Why does this have the syntax `:pathname`? Thats only needed if we wish to access `pathname` from the path, which I doubt we need to.

Reply:
react-router doesn't seem to understand the regex without `:param(regex)`.